dex/pair/src/fee.rs

```diff
@@ -256,32 +217,27 @@ pub trait FeeModule:
         available_amount: &BigUint,
         requested_token: &TokenIdentifier,
         destination_address: &ManagedAddress,
-    ) -> bool {
+    ) {
         let pair_address = self.get_extern_swap_pair_address(available_token, requested_token);
+
         let mut arg_buffer = ManagedArgBuffer::new_empty();
         arg_buffer.push_arg(requested_token);
         arg_buffer.push_arg(destination_address);
-        let result = self.raw_vm_api().direct_esdt_execute(
-            &pair_address,
-            available_token,
-            available_amount,
-            self.extern_swap_gas_limit().get(),
-            &ManagedBuffer::from(SWAP_NO_FEE_AND_FORWARD_FUNC_NAME),
-            &arg_buffer,
-        );
-
-        match result {
-            Result::Ok(()) => true,
-            Result::Err(_) => false,
-        }
+
+        self.raw_vm_api()
+            .direct_esdt_execute(
+                &pair_address,
+                available_token,
+                available_amount,
+                self.extern_swap_gas_limit().get(),
+                &ManagedBuffer::from(SWAP_NO_FEE_AND_FORWARD_FUNC_NAME),
+                &arg_buffer,
+            )
+            .unwrap();
```

Comment:
I suggest using a proxy for this, even if written by hand. Much easier to maintain.

Reply:
It adds quite a bit of bytecode to the wasm file but ok